x/stakeibc/keeper/icqcallbacks_validator_exchange_rate.go

```diff
@@ -76,18 +65,91 @@ func ValidatorExchangeRateCallback(k Keeper, ctx sdk.Context, args []byte, query
 	//     exchange_rate = num_tokens / num_shares
 	//  We can use `validator.TokensFromShares`, plug in 1.0 for the number of shares,
 	//    and the returned number of tokens will be equal to the internal exchange rate
-	validator.InternalExchangeRate = &types.ValidatorExchangeRate{
-		InternalTokensToSharesRate: queriedValidator.TokensFromShares(sdk.NewDec(1.0)),
-		EpochNumber:                strideEpochTracker.GetEpochNumber(),
-	}
+	currentSharesToTokensRate := queriedValidator.TokensFromShares(sdk.NewDec(1.0))
+	validator.InternalShareToTokensRate = currentSharesToTokensRate
 	hostZone.Validators[valIndex] = &validator
 	k.SetHostZone(ctx, hostZone)
 
-	k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_Validator, "Validator Internal Exchange Rate: %v",
-		validator.InternalExchangeRate.InternalTokensToSharesRate))
+	// Check if the validator was slashed by comparing the exchange rate from the query
+	// with the preivously stored exchange rate
+	validatorWasSlashed := false
+	if previousSharesToTokensRate.IsNil() || previousSharesToTokensRate.IsZero() {
+		k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_Validator,
+			"Previous Validator Exchange Rate Unknown"))
+
+	} else {
+		validatorWasSlashed = !previousSharesToTokensRate.Equal(currentSharesToTokensRate)
```

Comment:
nit: can we check L.T. and throw an error if G.T.? helps reduce our set of assumptions about how the exch rate works

Reply:
hmm that's a good question. I think if the exchange rate did increase, that definitely breaks my mental model, but at the same time it feels like we would still want to adjust the record keeping

Reply:
Yeah that's fair. We could keep the code as is and add a log + event emission if the exch rate increased just in case